.github/workflows/_helm-e2e.yaml

```diff
@@ -127,6 +134,40 @@ jobs:
                 ROLLOUT_TIMEOUT_SECONDS=900s
             fi
           done
+          
+          # Prepare external LLM arguments for external-llm-values.yaml files
+          external_llm_args=""
+          if [[ "${value_file}" == *"external-llm"* ]]; then
+            # Chart-specific mapping for environment variables (use LLM_SERVER_HOST_IP for all)
+            if [[ "${{ env.CHART_NAME }}" == "chatqna" ]]; then
+              if [[ -n "${{ env.LLM_SERVER_HOST_IP }}" ]]; then
+                external_llm_args="$external_llm_args --set externalLLM.LLM_SERVER_HOST_IP=${{ env.LLM_SERVER_HOST_IP }}"
+              fi
+              if [[ -n "${{ env.LLM_SERVER_PORT }}" ]]; then
+                external_llm_args="$external_llm_args --set externalLLM.LLM_SERVER_PORT=${{ env.LLM_SERVER_PORT }}"
+              fi
+            else
+              # DocSum/CodeGen use LLM_SERVICE_HOST_IP but we map from LLM_SERVER_HOST_IP
```

Comment:
Setting extra Helm variables that some Charts may ignore is fine, so per-chart checks for those can be removed.

But couldn't all applications (be changed to) use same env vars?

Reply:
The presence of `OPENAI_API_KEY` env variable triggers the external endpoint path in GenAIExamples afaik.